### PR TITLE
added console output in models and corresponding views.py changes

### PR DESCRIPTION
**Pull request by manasdadheech**

### CODE REVIEW

**Comment by avinassh** [comment]
http://chris.beams.io/posts/git-commit/

**Comment by kracekumar** [comment]
`requirements.txt` is missing.

File: mvp_cd/deploy_app/models.py [hunk 1]
```diff
@@ -3,21 +3,24 @@
...
-    is_success = models.BooleanField(default=False)
 
-
-class ProductionBuildInfo(models.Model):
-    curr_build_info = models.TextField()
-    is_success = models.BooleanField(default=False)
+class BuildInfo(models.Model):
+    circleci_json = models.TextField()
+    branch = models.CharField(max_length=255)
+    commitor = models.CharField(max_length=255)
```
Comment (kracekumar) [inline mvp_cd/deploy_app/models.py]: Correct spelling is `committer`.

File: mvp_cd/deploy_app/models.py [hunk 2]
```diff
@@ -3,21 +3,24 @@
...
+    circleci_url = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    vcs_revision = models.TextField(default='0000000')
+    deployment_status = models.BooleanField(default=False)
 
 
-class ProductionWorkerBuildInfo(models.Model):
-    curr_build_info = models.TextField()
+class ServerBuildInfo(models.Model):
+    ip = models.CharField(max_length=255)
```
Comment (kracekumar) [inline mvp_cd/deploy_app/models.py]: Have a look at https://docs.djangoproject.com/en/1.9/ref/models/fields/#genericipaddressfield

File: mvp_cd/deploy_app/models.py [hunk 3]
```diff
@@ -3,21 +3,24 @@
...
+    server_type = models.CharField(max_length=255)
```
Comment (kracekumar) [inline mvp_cd/deploy_app/models.py:23]: Have a look at `django Enum library https://pypi.python.org/pypi/django-enumfield/1.3b2`.

File: mvp_cd/deploy_app/models.py [hunk 4]
```diff
@@ -3,21 +3,24 @@
...
-class ProductionBuildInfo(models.Model):
-    curr_build_info = models.TextField()
-    is_success = models.BooleanField(default=False)
+class BuildInfo(models.Model):
+    circleci_json = models.TextField()
+    branch = models.CharField(max_length=255)
+    commitor = models.CharField(max_length=255)
+    circleci_url = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    vcs_revision = models.TextField(default='0000000')
```
Comment (kracekumar) [inline mvp_cd/deploy_app/models.py]: TextField is used for storing large data. Won't `varchar` be sufficient here ? Also default doesn't make sense.

File: mvp_cd/deploy_app/models.py [hunk 5]
```diff
@@ -3,21 +3,24 @@
...
 
-
-class ProductionBuildInfo(models.Model):
-    curr_build_info = models.TextField()
-    is_success = models.BooleanField(default=False)
+class BuildInfo(models.Model):
+    circleci_json = models.TextField()
+    branch = models.CharField(max_length=255)
+    commitor = models.CharField(max_length=255)
+    circleci_url = models.CharField(max_length=255)
```
Comment (kracekumar) [inline mvp_cd/deploy_app/models.py]: Use https://docs.djangoproject.com/en/1.9/ref/models/fields/#urlfield

File: mvp_cd/deploy_app/views.py [hunk 6]
```diff
@@ -1,118 +1,159 @@
...
-        build_info_item = StagingWorkerBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='False')
+        build_info_item = ServerBuildInfo(
+            is_success=False,
+            console_output=process_output,
+            ip=ip,
+            server_type=server_type,
+            build_info=build_info_obj)
+        build_info_obj.deployment_status = false
```
Comment (kracekumar) [inline mvp_cd/deploy_app/views.py]: Python doesn't have `false`, right one is `False`.

File: mvp_cd/deploy_app/views.py [hunk 7]
```diff
@@ -1,118 +1,159 @@
...
+        process_status(
+            request,
+            process_output, 0, ip, ServerType.APP_SERVER, build_info_obj)
+    except subprocess.CalledProcessError as e:
+        process_status(
+            request, e.output, 1, ip, ServerType.APP_SERVER, build_info_obj)
+
+
+def process_status(
+        request,
```
Comment (kracekumar) [inline mvp_cd/deploy_app/views.py]: Have a look at PEP 8 https://www.python.org/dev/peps/pep-0008/

File: mvp_cd/deploy_app/views.py [hunk 8]
```diff
@@ -1,118 +1,159 @@
...
+        process_status(
+            request,
+            process_output, 0, ip, ServerType.APP_SERVER, build_info_obj)
+    except subprocess.CalledProcessError as e:
+        process_status(
+            request, e.output, 1, ip, ServerType.APP_SERVER, build_info_obj)
+
+
+def deploy_app_production(
+        request, ip, pem_path, deployment_path, build_info_obj, vcs_revision):
```
Comment (kracekumar) [inline mvp_cd/deploy_app/views.py]: Follow consistent parameter placing for all functions.

File: mvp_cd/deploy_app/views.py [hunk 9]
```diff
@@ -1,118 +1,159 @@
...
+def build_detail_view(request, id):
+    build_info = get_object_or_404(BuildInfo, id=id)
+    serverbuildinfo_list = build_info.serverbuildinfo_set.all().order_by('-id')
+    return render_to_response(
+        'build_info.html',
+        {'build_info_item': build_info,
+            'serverbuildinfo_list': serverbuildinfo_list})
+
+
+def deploy_celery_staging(
```
Comment (kracekumar) [inline mvp_cd/deploy_app/views.py:71]: `deploy_app_production, deploy_app_staging, deploy_celery_production` can be written as single functions `deploy`. Variations can be passed as parameters.

File: mvp_cd/deploy_app/views.py [hunk 10]
```diff
@@ -1,118 +1,159 @@
...
 
 
 @api_view(['POST'])
-def deploy(request):
+def deploy_view(request):
     if request.data['payload']['outcome'] == 'success':
+        if request.data['payload']['branch'] not in ALLOWED_BRANCHES:
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        vcs_revision = request.data['payload']['vcs_revision']
```
Comment (kracekumar) [inline mvp_cd/deploy_app/views.py:20]: Incoming data needs to be validated for type and security. User data can't be trusted.

File: mvp_cd/deploy_app/templates/build_list.html [hunk 11]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
```
Comment (kracekumar) [inline mvp_cd/deploy_app/templates/build_list.html:5]: HTML is repeated in both places. Learn Django template inheritance. https://docs.djangoproject.com/en/1.9/ref/templates/language/#templates. Look at extending base template.

File: mvp_cd/deploy_app/templates/build_info.html [hunk 12]
```diff
@@ -2,21 +2,76 @@
...
+                {{ list_item.is_success }}</td>
+                </tr>
+                </table>
+                </div>
+                </div>
+                </div>
 
-    <h3> {{ build_info_item.curr_build_info }}</h3>
+                <H3 ALIGN='CENTER'>Console Output</H3>
+              <div><pre style="background:black;color:white;"> <code> {{ list_item.console_output }} </code></pre>
```
Comment (kracekumar) [inline mvp_cd/deploy_app/templates/build_info.html:65]: No inline CSS.

File: deploy_staging.sh [hunk 13]
```diff
@@ -1,6 +1,6 @@
 source $(dirname $0)/deploy.sh
```
Comment (kracekumar) [inline deploy_staging.sh:1]: This file is same as `deploy_prod.sh`. Two files can be merged and settings variable can be set by inspecting appropriate environment variable.

File: mvp_cd/deploy_app/models.py (hunk 1 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/models.py]: done

File: mvp_cd/deploy_app/models.py (hunk 2 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/models.py]: done

File: mvp_cd/deploy_app/models.py (hunk 4 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/models.py]: done

File: mvp_cd/deploy_app/views.py (hunk 6 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/views.py]: oops.. done

File: mvp_cd/deploy_app/models.py (hunk 5 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/models.py]: done

File: mvp_cd/deploy_app/views.py (hunk 7 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/views.py]: did read it and tried to make all the changes i could see. @kracekumar

File: mvp_cd/deploy_app/views.py (hunk 8 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/views.py]: done

File: mvp_cd/deploy_app/views.py (hunk 10 above)
Comment (manasdadheech) [inline mvp_cd/deploy_app/views.py:20]: It is coming from circleci and before that from github ? is their a chance for this to be bad too?

**Pull request merged**